src/components/v2v3/shared/FundingCycleConfigurationDrawers/NftDrawer/index.tsx

```diff
@@ -156,32 +240,45 @@ export default function NftDrawer({
     index: number
     newRewardTier: NftRewardTier
   }) => {
-    if (!tiersEqual({ tier1: newRewardTier, tier2: rewardTiers[index] })) {
+    if (
+      rewardTiers &&
+      !tiersEqual({ tier1: newRewardTier, tier2: rewardTiers[index] })
+    ) {
       setFormUpdated(true)
+      editingRewardTierIDsPush(rewardTiers[index].id)
     }
-    const newRewardTiers = rewardTiers.map((tier, i) =>
-      i === index
-        ? {
-            ...tier,
-            ...newRewardTier,
-          }
-        : tier,
-    )
+    const newRewardTiers = rewardTiers
```

Comment:
**suggestion**: maybe an alternative here is to use a key-value dict rather than array, of the following form:

{
   `<tierId>`: <tier>
}

This should remove the need for complicated upsert logic like this i think.

Reply:
Problem is tiers don't have ID's before they're launched. So I don't see how this would work when projects without NFTs go to launch a new collection through this reconfiguration flow.